### PR TITLE
SI-5730 hide constructors of sealed abstract classes in scaladoc

### PR DESCRIPTION
**Pull request by gourlaysama**
Sealed abstract classes (like `List`) have a primary constructor, public
by default. It can never be called by external code but it shows up in
the scaladoc as a nice `new List()` construtor...

If a class is only abstract, the constructor is still useful because
people can subclass and call it. If it is only sealed (i.e. effectively final),
then it is the normal constructor of a final class. But sealed _and_
abstract makes documenting the constructor useless.

This should remove the misleading constructors of `List`, `Double`,
`Option` and others from the scaladoc.

### CODE REVIEW

File: test/scaladoc/run/t5730.check [hunk 1]
```diff
@@ -0,0 +1 @@
+Done.
```
Comment (som-snytt) [inline test/scaladoc/run/t5730.check:1]: FWIW, I was going to make the "done" message easy to override (and not emit).  If partest doesn't hang and scaladoc doesn't hang, the check file doesn't add much.

https://github.com/som-snytt/scala/blob/ccdf22032616a72ae683d3c2145e536e117dfc97/src/scaladoc/scala/tools/partest/ScaladocModelTest.scala#L81

https://github.com/scala/scala/pull/3353/files

**Comment by gourlaysama** [comment]
Now that the tests pass (and took only 5h26min...): review @VladUreche :)

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 2]
```diff
@@ -753,8 +753,10 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
         })
       }
       else if (bSym.isConstructor)
```
Comment (VladUreche) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala:755]: @gourlaysama, I don't understand why we would show constructors at all. I would say `if (bSym.isConstructor) None` and that's it. Right?

Comment (VladUreche) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala:755]: Now I understand what you mean. LGTM!

**Comment by VladUreche** [comment]
LGTM!

**Comment by VladUreche** [comment]
Thanks @gourlaysama!

**Pull request merged**